app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_secure_password
+  validates :username, uniqueness: true
+  validates :email, uniqueness: true
```

Comment:
Looking for uniqueness 👍 
Could extend this making sure that a correct format for emails is used ...

Reply:
What would happen if the user did not enter a user name or password when first signing up? Would it still work?

Reply:
Could also look at length